src/thread_task.rs

```diff
@@ -999,24 +988,25 @@ pub unsafe extern "C" fn rav1d_worker_task(data: *mut c_void) -> *mut c_void {
             }
             park(tc, ttd);
             continue 'outer;
-        }
-
+        };
         // found:
         // remove t from list
-        if !prev_t.is_null() {
-            (*prev_t).next = (*t).next;
+        if let Some(prev_t) = prev_t.as_deref_mut() {
+            prev_t.next = t.next;
         } else {
-            (*f).task_thread.task_head = (*t).next;
+            f.task_thread.task_head = t.next;
         }
-        if ((*t).next).is_null() {
-            (*f).task_thread.task_tail = prev_t;
+        if (t.next).is_null() {
+            f.task_thread.task_tail = prev_t
+                .as_deref_mut()
+                .map_or(ptr::null_mut(), |p| p as *mut Rav1dTask);
```

Comment:
Why? Same difference here.

Reply:
It'll be easy to search and replace in the future since we use `.map_or_else(ptr::null_mut,` lots of other places.  Also I think clippy will suggest this but that's not really important.